### ISSUE TITLE
Clarification on Initial Steps: -Data Preprocessing or Model Development

### ISSUE THREAD

**Issue by zidan18Ahd** [malariagen/gsoc-mosquito-taxon-classifier#4]
Hi Team,
I am going to train the taxon classifier on genotype (GT) data (features.tsv) and its respective taxon labels (labels.tsv), as a result I would like to confirm the suggested first steps:
1)Should I develop the model in the first instance, employing the current **random forest classifier as the baseline?** or use some other alternate ways of modelling ?
2)Or do we emphasize preprocessing in the form of data validation, addressing imbalanced classes?

Your advice on the best method to guarantee efficiency and robustness would be appreciated immensely. 
Hope to hear from you.
Zidan Ahmed
cc: @ahernank .

**Comment by hanumanihar** [comment]
Sir, since the worry is about efficiency and speed...other models like XGBoost (to my knowledge this could better address the imbalanced data ) and maybe LightGBM . If preprocessing is not going to be prioritized, then maybe a hybrid of the existing one with XGBoost would help? could you please share your initial goals and tradeoffs that led to current choice of using a random forest model?

**Comment by hanumanihar** [comment]
Hello everyone,
I just have a small doubt ... Is this going to be a team project or a solo thing? what I mean to ask is, once the proposals are reviewed, how many members per project are you looking for? is it something like one person for data preprocessing, machine learning part and another for the cloud deployment part? or are we encouraged to do the entire thing to maintain consistency?

**Comment by jonbrenas** [comment]
Thank you @zidan18Ahd and @hanumanihar.

You are free to choose whichever method you think would work best. The expected deliverable is a taxon-classifier so  preprocessing will not suffice on its own but, depending on how you want to approach the problem, it might be a very important part of your proposal.

We are only able to select one candidate for this project but, because there are many ways to tackle the issue and the problem is important, we also hope to create a community that would be able to, in the short term, solve this problem and, in the long term, solve various other similar problems using ML.
